static/src/javascripts/projects/commercial/modules/prebid/appnexus.js

```diff
@@ -0,0 +1,136 @@
+// @flow strict
+
+import config from 'lib/config';
+import {
+    buildPageTargeting,
+    buildAppNexusTargetingObject,
+} from 'common/modules/commercial/build-page-targeting';
+
+import {
+    getLargestSize,
+    containsLeaderboard,
+    containsLeaderboardOrBillboard,
+    containsMpu,
+    containsMpuOrDmpu,
+    getBreakpointKey,
+} from './utils';
+
+import type { PrebidAppNexusParams, PrebidSize } from './types';
+
+const getAppNexusInvCode = (sizes: Array<PrebidSize>): ?string => {
+    const device: string = getBreakpointKey() === 'M' ? 'M' : 'D';
+    const section: string = config.get('page.section', 'unknown');
+    const slotSize: PrebidSize | null = getLargestSize(sizes);
+    if (slotSize) {
+        return `${device}${section.toLowerCase()}${slotSize.join('x')}`;
+    }
+};
+
+export const getAppNexusPlacementId = (sizes: PrebidSize[]): string => {
+    const defaultPlacementId: string = '13915593';
+    switch (config.get('page.edition')) {
+        case 'UK':
+            switch (getBreakpointKey()) {
+                case 'D':
+                    if (containsMpuOrDmpu(sizes)) {
+                        return '13366606';
+                    }
+                    if (containsLeaderboardOrBillboard(sizes)) {
+                        return '13366615';
+                    }
+                    return defaultPlacementId;
+                case 'M':
+                    if (containsMpu(sizes)) {
+                        return '13366904';
+                    }
+                    return defaultPlacementId;
+                case 'T':
+                    if (containsMpu(sizes)) {
+                        return '13366913';
+                    }
+                    if (containsLeaderboard(sizes)) {
+                        return '13366916';
+                    }
+                    return defaultPlacementId;
+                default:
+                    return defaultPlacementId;
+            }
+        default:
+            return defaultPlacementId;
+    }
+};
```

Comment:
This block will need updating like this: https://github.com/guardian/frontend/pull/20531